src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -3919,5 +3919,1172 @@ class Test1 : I1
+                // (2,15): error CS8502: 'I1.E7.add' cannot implement interface member 'I1.E7.add' in type 'Test2' because the target runtime doesn't support default interface implementation.
+                // class Test2 : I1
+                Diagnostic(ErrorCode.ERR_RuntimeDoesNotSupportDefaultInterfaceImplementationForMember, "I1").WithArguments("I1.E7.add", "I1.E7.add", "Test2")
+                );
+
+            ValidateEventImplementation_501(compilation3.SourceModule, "Test2");
+        }
+
+        [Fact]
+        public void EventImplementation_701()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    event System.Action E7 
+    { 
+        add {} 
+        remove {} 
+    }
+}
+
+class Test1 : I1
+{}
+";
+            var compilation1 = CreateCompilationWithMscorlib(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp7));
+            Assert.True(compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+
+            compilation1.VerifyDiagnostics(
+                // (6,9): error CS8107: Feature 'default interface implementation' is not available in C# 7.  Please use language version 7.1 or greater.
+                //         add {} 
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7, "add").WithArguments("default interface implementation", "7.1").WithLocation(6, 9),
+                // (7,9): error CS8107: Feature 'default interface implementation' is not available in C# 7.  Please use language version 7.1 or greater.
+                //         remove {} 
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7, "remove").WithArguments("default interface implementation", "7.1").WithLocation(7, 9)
+                );
+
+            ValidateEventImplementation_501(compilation1.SourceModule, "Test1");
+
+            var source2 =
+@"
+class Test2 : I1
+{}
+";
+
+            var compilation2 = CreateCompilationWithMscorlib(source2, new[] { compilation1.ToMetadataReference() }, options: TestOptions.DebugDll,
+                                                            parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp7));
+            Assert.True(compilation2.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            compilation2.VerifyDiagnostics();
+
+            ValidateEventImplementation_501(compilation2.SourceModule, "Test2");
+
+            CompileAndVerify(compilation2, verify: false,
+                symbolValidator: (m) =>
+                {
+                    var test2Result = (PENamedTypeSymbol)m.GlobalNamespace.GetTypeMember("Test2");
+                    Assert.Equal("I1", test2Result.Interfaces.Single().ToTestDisplayString());
+                    ValidateEventImplementation_501(m, "Test2");
+                });
+        }
```

Comment:
I plan to cover them eventually, tracked in https://github.com/dotnet/roslyn/issues/17952.